app/models/concerns/specimen_role.rb

```diff
@@ -9,19 +9,33 @@ module SpecimenRole
 
   end
 
+  def is_quality_control
+    specimen_role == 'q'
+  end
+
   class_methods do
+    def specimen_roles
+      @@specimen_roles ||= build_specimen_roles
```

Comment:
Be careful with class variables `@@` (vs class instance variables `@`) in concerns. It can quickly bite back. Yes, classes themselves are instances (of `Class`) and they can have both class variables (shared with descendant classes) _and_ instance variables :exploding_head: (private to the class).

I think it should work here, since its within a`class_methods` block that should inject the methods right into the class (so the variables shouldn't be shared by the module to different classes), but... I'm always wary of them.

Reply:
But maybe the block merely creates a `ClassMethods` module and extends the class with it, in which case the class variables will bite:

once specimen roles are loaded for Sample or Batch they will be used for the other one :skull:

Reply:
Great suggestion! I'll fix it 👍